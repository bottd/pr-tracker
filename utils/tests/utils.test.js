const utils = require('../utils');

describe('utils', () => {
  describe('parseLinks', () => {
    it('Should should return an object with <links> assigned to their rel', () => {
      const expected = {
        test: 'https://api.github.com',
      };
      const links = '<https://api.github.com>; rel="test"';
      const result = utils.parseLinks(links);
      expect(result).toEqual(expected);
    });
    it('Should work with multiple links', () => {
      const expected = {
        next:
          'https://api.github.com/repositories/10270250/pulls?state=open&page=2',
        last:
          'https://api.github.com/repositories/10270250/pulls?state=open&page=5',
      };
      const links =
        '<https://api.github.com/repositories/10270250/pulls?state=open&page=2>; rel="next", <https://api.github.com/repositories/10270250/pulls?state=open&page=5>; rel="last"';
      const result = utils.parseLinks(links);
      expect(result).toEqual(expected);
    });
  });
});
