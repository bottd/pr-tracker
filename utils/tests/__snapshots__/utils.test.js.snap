// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils fetchAllPulls Should return all pulls in one array 1`] = `
Array [
  Object {
    "comments_url": "example.com/comments/1",
    "commits_url": "example.com/commits/1",
    "extraKey": "Should be thrown out",
    "title": "A Pull Request",
    "user": Object {
      "login": "bottd",
    },
  },
  Object {
    "comments_url": "example.com/comments/2",
    "commits_url": "example.com/commits/2",
    "extraKey": "Should be thrown out",
    "title": "Another Pull Request",
    "user": Object {
      "login": "bottd",
    },
  },
]
`;
